openhtf/__init__.py

```diff
@@ -105,31 +118,79 @@ def code_info(self):
   def metadata(self):
     return self._test_info.metadata
 
-  def Execute(self, loop=None,
-              test_start=triggers.AutoStart, test_stop=triggers.AutoStop,
-              http_port=http_api.DEFAULT_HTTP_PORT):
+  def Configure(self, **kwargs):
+    """Update test-wide configuration options.
+
+    Valid kwargs:
+      http_port: Port on which to run the http_api, or None to disable.
+      output_callbacks: List of output callbacks to run, typically it's better
+          to use AddOutputCallbacks(), but you can pass [] here to reset them.
+    """
+    for key, value in kwargs.iteritems():
+      setattr(self._test_options, key, value)
+
+  def Stop(self, *_):
+    _LOG.error('Received SIGINT.')
+    with self._lock:
+      if self._executor:
+        # TestState str()'s nicely to a descriptive string, so let's log that
+        # just for good measure.
+        _LOG.error('Stopping Test due to SIGINT: %s', self._executor.GetState())
+        self._executor.Stop()
+      else:
+        # Once we're stopped for SIGINT, don't try to start anything else.
+        _LOG.error('No Test running, ignoring SIGINT.')
```

Comment:
This only happens (because of the else) if SIGINT comes before `Execute()` gets a few lines in. After that, Ctrl-C won't stop future executions, is that what you want?